runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -4,13 +4,32 @@ local ms = require('vim.lsp.protocol').Methods
 local api = vim.api
 local M = {}
 
----@class (private) vim.lsp.inlay_hint.bufstate
+---@class (private) vim.lsp.inlay_hint.globalstate Global state for inlay hints
+---@field enabled boolean Whether inlay hints are enabled for this scope
```

Comment:
Is the scope always the buffer?

Reply:
It could be enabled globally because we do not only expect to be enabled on buffers that exist now but also for the buffers that will be created in the future.